### PR TITLE
Bundle dependency list ui redesign.

### LINKED ISSUE #4182: Optimize bundle detail dependency list UI
### Overview
In the bundle detail sidebar, dependencies with longer names are currently rendered in a way that is unpleasing to the eye. For example here is the dependencies list in [this worksheet](https://codalab.stanford.edu/bundles/0x3b15e291c825412680540b1805716627):

Screen Shot 2022-08-03 at 3 47 03 PM

### High-Level Goal
The goal of this issue is to explore the best way to render a list of bundle dependencies, and then implement your solution. UI options should be presented to the team prior to implementation.

### PR DESCRIPTION
**Pull request by leilenah**
### Reasons for making this change

The current bundle dependency list UI doesn't make great usage of the limited space available in the bundle detail sidebar:

Screen Shot 2022-08-03 at 3 47 03 PM

The reason for this change is to only show the user what they need to know about each dependency in the most clean and concise way possible.

### Related issues

https://github.com/codalab/codalab-worksheets/issues/4182
https://github.com/codalab/codalab-worksheets/issues/4192

### Screenshots

image

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [ ] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [ ] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by percyliang** [comment]
Thanks!  Is it possible to use all that whitespace under 'Dependencies'?  It does break the key/value pattern, but dependencies are sort of special enough that we might want to have them look different anyway.

In a way, like `command`, it is central to defining the run, which is moved out.

**Comment by leilenah** [comment]
@percyliang this is what I'm thinking for the dependency list UI. Plz lmk your thoughts.

I also want to update `Allow Failed`, as well as the other boolean fields to use a toggle. I have created a separate issue for this https://github.com/codalab/codalab-worksheets/issues/4185

**Comment by leilenah** [comment]
@percyliang that's def possible -- will prepare something to show you

**Comment by leilenah** [comment]
cc @pranavjain in case you have notes

**Comment by leilenah** [comment]
cc @epicfaace

File: docs/REST-API-Reference.md [hunk 1]
```diff
@@ -196,6 +196,7 @@ Name | Type
 `parent_uuid` | String
 `parent_path` | String
 `parent_name` | Method
+`parent_state` | Method
```
Comment (epicfaace) [inline docs/REST-API-Reference.md:199]: What does "Method" even mean? How can a REST API return a "method"?

Comment (epicfaace) [reply docs/REST-API-Reference.md:199]: Shouldn't this just be a "string" type (and similarly for parent_name)?

File: codalab/rest/schemas.py [hunk 2]
```diff
@@ -155,11 +155,19 @@ class BundleDependencySchema(PlainSchema):
...
 
     def get_parent_name(self, dep):
         uuid = dep['parent_uuid']
         return local.model.get_bundle_names([uuid]).get(uuid)
 
+    def get_parent_state(self, dep):
+        uuid = dep['parent_uuid']
+        try:
+            return local.model.get_bundle_state(uuid)
+        except NotFoundError:
```
Comment (epicfaace) [inline codalab/rest/schemas.py:168]: when would NotFoundError be thrown? Do we need this try-except?

**Review by epicfaace — APPROVED** [review]

File: docs/REST-API-Reference.md (hunk 1 above)
Comment (leilenah) [reply docs/REST-API-Reference.md:199]: @epicfaace This was auto-generated. I think it signifies that some logic runs in the schema to get this value (specifically `get_parent_name()` and `get_parent_state()`.

Comment (leilenah) [reply docs/REST-API-Reference.md:199]: String probably makes more sense in terms of the docs tho....not sure how these are generated

File: codalab/rest/schemas.py (hunk 2 above)
Comment (leilenah) [reply codalab/rest/schemas.py:168]: @epicfaace I observed an edge case in which `NotFoundError` is thrown [here](https://github.com/codalab/codalab-worksheets/blob/master/codalab/model/bundle_model.py#L1252) when we try to fetch a state for dependencies that were force-deleted.

**Pull request merged**